% !TEX root = report.tex

\chapter{Overview} \label{ch:overview}

Traditional methods of predicting residue residue contacts. These advances have largely been driven by the use of residue coevolution, deduced from multiple sequence alignments, in order to predict interactions. However, their reliance on evolutionary information precludes the application of these approaches to the design of entirely new proteins, with no existing homologs found in nature.

\noindent\makebox[\linewidth]{\rule{\paperwidth}{0.4pt}}

There is strong interest in new tools and approaches for improving protein stability and increasing the affinity of proteins to their targets. In academic research laboratories, such tools could be used to design proteins that are easier to crystallize and that show higher yield and are less prone to aggregation when expressed in heterologous systems (\cite{DellerProteinstabilitycrystallographer2016,PopovComputationaldesignthermostabilizing2018}). In industrial process engineering, such tools could be used to design biocatalysts which remain active in inhospitable conditions, including high temperatures and pressures, acidic or basic pH, different salt types and concentrations, and admixture of organic solvents (\cite{BommariusStabilizingbiocatalysts2013}). In the pharmaceutical industry, such tools could be used to optimize protein and peptide ``biologics'' to improve their shelf life and increase their affinity and specificity to desired targets (\cite{DavilaCD19TargetedCARCells2016}). 

Existing approaches for protein optimization and design are either too computationally expensive, too inaccurate, or both, to be practical for routine use. Alchemical free energy calculations based on molecular dynamics can accurately predict the change in the Gibbs free energy of folding or binding ($\Delta \Delta G$) associated with mutations \cite{SeeligerGeometryBasedSamplingConformational2007,GapsysAccurateRigorousPrediction2016}. However, evaluating the impact of a single mutation can take hours to days on a modern graphical processing unit, making this approach intractable for protein design. Statistical potentials can be used to evaluate a large number of amino acid substitutions proposed by Markov chain Monte Carlo or other sampling techniques \cite{KuhlmanDesignNovelGlobular2003,SchymkowitzFoldXwebserver2005}. However, statistical potentials are not very accurate, and the sampling procedure generally keeps the backbone of the protein fixed and thereby only selects sequences around a local energy minimum of a single protein conformation. To date, most successful examples of computational protein design have involved a final step where hundreds of thousands of designed sequences are screened using high-throughput experimental techniques to select a handful of sequences that produce proteins with notably improved properties \cite{SunProteinengineeringhighly2016,RocklinGlobalanalysisprotein2017}.

The overarching goal of my PhD project is to develop computational methods that can be used to construct libraries of proteins that have higher stability, higher catalytic activity, or higher affinity to their target than the starting structure. Two recent developments can help me in achieving this goal. First, advances in next-generation sequencing technology have led to enormous growth in the amount of genomic data that is available, and this data has already been shown to be useful in structural biology. For example, in our work on ELASPIC \cite{WitvlietELASPICwebserverproteomewide2016}, we observed that using a mutation deleteriousness score in addition to structural information can improve the accuracy with which we can predict the energetic effect of mutations. As another example, co-evolutionary information extracted from multiple sequence alignments has been used to accurately predict the three-dimensional structure of proteins \cite{HopfThreedimensionalstructuresmembrane2012,OvchinnikovProteinstructuredetermination2017}, suggesting that this data can also be immensely useful for protein design. Second, advances in neural networks, coupled with the growth in the volume of data and the computational resources that are available, have led to striking breakthroughs in many fields such as image detection and natural language processing. While there has been some work on using neural networks for protein optimization and design \cite{WangComputationalProteinDesign2018,AlQuraishiEndtoenddifferentiablelearning2018}, this application remains relatively under-explored and much headway remains to be made.

The report is structured as follows. In Chapter \ref{ch:article1}, we discuss how molecular dynamics simulations can be used to evaluate the energetic impact of mutations on protein folding and binding. In Chapter \ref{ch:article2}, we discuss the development and optimization of statistical potentials and the application of statistical potentials to protein design. In Chapter \ref{ch:article4}, we examine a study which uses recurrent neural networks, trained on a dataset of sequential and structural information, to ``fold'' proteins into their three-dimensional shape. In Chapter \ref{ch:article5}, we examine a study which uses generative adversarial neural networks to optimize existing DNA sequences and generate new DNA sequences with desirable properties. Finally, in Chapter \ref{ch:my_project}, we outline our progress in training a neural network to predict whether a given sequence matches a given adjacency matrix. We also discuss the difficulties that we are facing in training a generative adversarial network to generate sequences that fold into a given shape.
